### PR TITLE
Add `Interaction::Released` (Detect the release of a UIButton)

### LINKED ISSUE #5769: Add support for detecting release of a button
## What problem does this solve or what need does it fill?

Currently we detect when a button is being hovered, or clicked, but not _released_.
```
pub enum Interaction {
    /// The node has been clicked
    Clicked,
    /// The node has been hovered over
    Hovered,
    /// Nothing has happened
    #[default]
    None,
}
```
In this case `Clicked` is true when the user presses down on the button.

## What solution would you like?

Add a `Released` variant to the `Interaction` enum, then add whatever logic necessary to correctly detect it.

## What alternative(s) have you considered?

- Not supporting it is not really an option. Most applications (and games) only trigger the button's action on release.

### PR DESCRIPTION
**Pull request by Weibye**
# Objective

Fix #5769

## Solution

Change to `Interaction::Released` instead of `Interaction::None` for 1 frame as the button is released.

---

## Added
- `Interaction` now be considered `Interaction::Released` for 1 frame when the user has released interaction on the button.

## Feedback Wanted / Unresolved Questions

This solution relies on the ui_focus system running for 1 more frame after the interaction has been released, meaning that it does not work currently work when using the "DesktopMode" winit setting that only runs systems when user-input has occurred. It will be stuck in "Released" state until some new input is provided by the user.

Any ideas as to how that can be solved?

### CODE REVIEW

File: crates/bevy_ui/src/focus.rs [hunk 1]
```diff
@@ -37,6 +37,8 @@ use smallvec::SmallVec;
 pub enum Interaction {
     /// The node has been clicked
     Clicked,
+    /// The node has been released from
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/focus.rs:40]:
```suggestion
    /// The node has just been released
```

**Comment by alice-i-cecile** [comment]
> This solution relies on the ui_focus system running for 1 more frame after the interaction has been released, meaning that it does not work currently work when using the "DesktopMode" winit setting that only runs systems when user-input has occurred. It will be stuck in "Released" state until some new input is provided by the user.

That's pretty unfortunate. We could add another system at the end of the frame to reset them? Or run for two frames on interaction in `DesktopMode`.

**Comment by Weibye** [comment]
> We could add another system at the end of the frame to reset them?

When does input reset their "just_pressed" / "just_released" states? Is it within the same frame but at the end of the update stage?

**Comment by alice-i-cecile** [comment]
> When does input reset their "just_pressed" / "just_released" states? Is it within the same frame but at the end of the update stage?

Near the start of each frame in `PreUpdate`. I actually think this is a bit of a mistake, as it wreaks havoc with things like fixed time steps, but that's the current strategy.

**Comment by alice-i-cecile** [comment]
This is relatively complex, and we haven't reached consensus on the design. Removing from the 0.10 milestone.

**Comment by shanecelis** [comment]
I thought I wanted `Released` too, but I was convinced that `Released` is a transition not a state by @Shatur. I added a [longer comment here](https://github.com/bevyengine/bevy/pull/10141#issuecomment-2241079564).

**Pull request closed without merging**